src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,134 @@
+---
+section: Guides
+chapter: Main Codebase
+title: Maintaining Subsystems
+description: How to maintain subsystems within the main codebase.
+slug: /guides/main/maintaining-subsystems
+---
+
+This page details how to maintain various subsystems within the main codebase.
+
+# Vision
+
+Before vision can work on the robot, odometry and kinematics must work on the robot. If these are inaccurate, then the green horizon will be inaccurate.
+
+## Dataset Generation
+
+Datasets for training the Visual Mesh can be found on the Network-Attached Storage (NAS). If you don't have access to the NAS, ask a team leader for an account. The NAS contains millions of NUpbr training images and large TensorFlow records containing the correct data to train the Visual Mesh on.
```

Comment:
This feels like the wrong place to be saying specifics about NAS access to me. I feel like this should ideally link to a page with information about the NAS and how to get access.

Reply:
If it seems strange, we can get rid of it. But I don't think it's worth having a completely separate page just about the NAS, there's not really a lot to say about it.

Reply:
How about something like this?

```suggestion
Synthetic and semi-synthetic training data for vision can be generated using NUpbr and NUgan. Pre-generated datasets for training the Visual Mesh can be found on the Network-Attached Storage (NAS) in the lab.
```

Reply:
That reads good.